unittest/gunit/keyring_vault/vault_io-t.cc

```diff
@@ -0,0 +1,680 @@
+  }
+
+  TEST_F(Vault_io_test, FlushAndRemoveSingleKey)
+  {
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    EXPECT_FALSE(vault_io.init(&credential_file_url));
+    Vault_key key("key1", "AES", "rob", "Robi", 4);
+    key.set_key_operation(STORE_KEY);
+    EXPECT_FALSE(vault_io.flush_to_storage(&key));
+    Vault_key key_to_remove(key);
+    key_to_remove.set_key_operation(REMOVE_KEY);
+    EXPECT_FALSE(vault_io.flush_to_storage(&key_to_remove));
+  }
+
+  TEST_F(Vault_io_test, FlushKeyRetrieveDeleteInit)
+  {
+    Vault_io vault_io(logger, vault_curl, vault_parser);
+    EXPECT_FALSE(vault_io.init(&credential_file_url));
+    Vault_key key("key1", "AES", "rob", "Robi", 4);
+    key.set_key_operation(STORE_KEY);
+    EXPECT_FALSE(vault_io.flush_to_storage(&key));
+    Vault_key key1_id("key1", NULL, "rob", NULL, 0);
+    EXPECT_FALSE(vault_io.retrieve_key_type_and_data(&key1_id));
+    EXPECT_STREQ(key1_id.get_key_signature()->c_str(), "4_key13_rob");
+    ASSERT_TRUE(memcmp(key1_id.get_key_data(), "Robi", key1_id.get_key_data_size()) == 0);
+    EXPECT_STREQ("AES", key1_id.get_key_type()->c_str());
+
+    Vault_key key_to_remove(key);
+    key_to_remove.set_key_operation(REMOVE_KEY);
+    EXPECT_FALSE(vault_io.flush_to_storage(&key_to_remove));
+
+    Vault_curl *vault_curl2 = new Vault_curl(logger);
+    Vault_parser *vault_parser2 = new Vault_parser(logger);
+    Vault_io vault_io2(logger, vault_curl2, vault_parser2);
+    EXPECT_FALSE(vault_io2.init(&credential_file_url));
+    ISerialized_object *serialized_keys= NULL;
+    EXPECT_FALSE(vault_io2.get_serialized_object(&serialized_keys));
+    ASSERT_TRUE(serialized_keys == NULL); //no keys
+  }
+
+  class Mock_vault_curl : public IVault_curl
+  {
+  public:
+    MOCK_METHOD1(init, bool(const Vault_credentials &vault_credentials));
+    MOCK_METHOD1(list_keys, bool(Secure_string *response));
+    MOCK_METHOD2(write_key, bool(const Vault_key &key, Secure_string *response));
+    MOCK_METHOD2(read_key, bool(const Vault_key &key, Secure_string *response));
+    MOCK_METHOD2(delete_key, bool(const Vault_key &key, Secure_string *response));
+  };
+
+  TEST_F(Vault_io_test, ErrorFromVaultCurlOnVaultIOInit)
+  {
+    delete vault_curl;
+    Mock_vault_curl *mock_curl = new Mock_vault_curl();
+    Vault_io vault_io(logger, mock_curl, vault_parser);
+
+    EXPECT_CALL(*mock_curl, init(_))
+      .WillOnce(Return(TRUE)); // init unsuccessfull
+    EXPECT_TRUE(vault_io.init(&credential_file_url));
+  }
+
+  TEST_F(Vault_io_test, ErrorFromVaultCurlOnListKeys)
+  {
+    delete vault_curl;
+    Mock_vault_curl *mock_curl = new Mock_vault_curl();
+    Vault_io vault_io(logger, mock_curl, vault_parser);
+
+    EXPECT_CALL(*mock_curl, init(_))
+      .WillOnce(Return(FALSE)); // init successfull
+    EXPECT_FALSE(vault_io.init(&credential_file_url));
+
+    ISerialized_object *serialized_object;
+
+    EXPECT_CALL(*mock_curl, list_keys(_))
+      .WillOnce(Return(TRUE)); //failed to list keys
```

Comment:
All mock methods are bool, not my_bool, hence lowercase constants everywhere. Space after //